src/main/kotlin/com/jetbrains/snakecharm/lang/parser/SmkStatementParsing.kt

```diff
@@ -279,23 +310,82 @@ class SmkStatementParsing(
 //    }
 
 
-    private fun parseIdentifier(): Boolean {
+    private fun parseIdentifier(type: IElementType = SmkElementTypes.REFERENCE_EXPRESSION): Boolean {
         val referenceMarker = myBuilder.mark()
         if (Parsing.isIdentifier(myBuilder)) {
             Parsing.advanceIdentifierLike(myBuilder)
-            referenceMarker.done(SmkElementTypes.REFERENCE_EXPRESSION)
+            referenceMarker.done(type)
             return true
         }
         referenceMarker.drop()
         return false
     }
+
+    /**
+     * Parsing 'use' statement. Starts after first identifier and ends before the colon
+     */
+    private fun parseUseDeclaration(): Boolean {
+        var hasImport = false // Does this section contains 'module' keyword
+        val listOfRules = myBuilder.tokenType == PyTokenTypes.MULT
+        if (!listOfRules && myBuilder.tokenType != PyTokenTypes.IDENTIFIER) { // No rule name, no '*'
+            myBuilder.error(PyPsiBundle.message("PARSE.expected.symbols.first.quotation", "*", "rule name"))
+        } else if (!listOfRules) { // Have rule name, so mark it as reference
+            parseIdentifier()
+        } else { // Have '*'
+            nextToken()
+        }
+
+        if (myBuilder.tokenType != PyTokenTypes.AS_KEYWORD) {
+            checkMatches(PyTokenTypes.FROM_KEYWORD, PyPsiBundle.message("PARSE.0.expected", "from"))
+
+            // Creates reference to module definition
+            if (!parseIdentifier(PyElementTypes.REFERENCE_EXPRESSION)) {
```

Comment:
Why we use Python reference expression, not our own (SmkElementTypes.REFERENCE_EXPRESSION) ?